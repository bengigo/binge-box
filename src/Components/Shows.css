.search-field {
  background-color: var(--dark);
  height: 160px;
  padding: 15px 5px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
}

.count {
  padding: 8px;
}

input {
  color: black;
  text-align: center;
}

.show-list {
  display: grid;
  grid: 1fr / auto auto;
  list-style-type: none;
}

.main-list:nth-child(4n + 3) {
  background-color: #3f62a4;
}

.main-list:nth-child(4n + 2) {
  background-color: #3f62a4;
}

.material-symbols-outlined {
  font-size: 40px;
}

a {
  text-decoration: none;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.main-poster {
  height: 150px;
}

.shows-info {
  display: flex;
  flex-direction: column;
  align-items: center;
}

h2 {
  text-align: center;
  font-size: 20px;
  width: 150px;
}

.icon {
  display: flex;
  justify-content: flex-end;
  width: 100%;
}
