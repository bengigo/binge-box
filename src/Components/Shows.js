import React, { useEffect, useState } from 'react';
import './Shows.css';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchShows } from '../Redux/shows';

const Shows = () => {
  const shows = useSelector((state) => state.shows);

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchShows());
  }, []);

  const [query, setQuery] = useState('');

  return (
    <div>
      <section className="search-field">
        <p className="count">
          {shows.length}
          {' '}
          SHOWS AVAILABLE
        </p>
        <input className="search-bar" placeholder="search by name" onChange={(event) => setQuery(event.target.value)} />
      </section>

      <section className="main-display">
        {shows.filter((show) => {
          if (query === '') {
            return true;
          } if (show.name.toLowerCase().includes(query.toLowerCase())) {
            return true;
          } return false;
        }).map((show) => (
          <article key={show.id}>
            <Link to={`/ShowDetails/${show.id}`} state={{ id: `${show.id}` }}>

              <p className="icon">
                <span className="material-symbols-outlined">
                  arrow_circle_right
                </span>
              </p>

              <h2 className="show-name">{show.name}</h2>
              <p className="rating">{show.rating}</p>
            </Link>
          </article>
        ))}
      </section>
    </div>
  );
};

export default Shows;
