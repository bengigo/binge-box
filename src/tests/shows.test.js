import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import store from '../Redux/ConfigureStore';
import Shows from '../Components/Shows';

describe('Shows', () => {
  it('should render shows component', () => {
    render(
      <Provider store={store}>
        <Shows />
      </Provider>,
    );
    expect(Shows).toMatchSnapshot();
  });
});
