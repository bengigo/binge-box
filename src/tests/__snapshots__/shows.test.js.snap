// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows should render shows component 1`] = `[Function]`;
